plugins/org.eclipse.osee.framework.core/src/org/eclipse/osee/framework/core/data/OrcsTokenServiceImpl.java

```diff
@@ -437,6 +437,40 @@ public Set<ArtifactTypeToken> getValidArtifactTypes(AttributeTypeToken attribute
       return artifactTypes;
    }
 
+   @Override
+   public List<RelationTypeToken> getValidRelationTypes() {
+      Collection<RelationTypeToken> relationTypes = getRelationTypes();
+      List<RelationTypeToken> invalidRels = new ArrayList<>();
+      //populate convertedRelations if empty
+      if (convertedRelations.isEmpty()) {
```

Comment:
Shouldn't this be done elsewhere?

Reply:
Its done the first time this is called OR the first time a specific relation is called out.